src/engine/renderer/tr_shade.cpp

```diff
@@ -34,12 +34,28 @@ This file deals with applying shaders to surface data in the tess struct.
 
 static void EnableAvailableFeatures()
 {
-	glConfig2.dynamicLight = r_dynamicLight->integer;
+	glConfig2.dynamicLight = r_dynamicLight.Get();
+	glConfig2.staticLight = r_staticLight.Get();
 
-	if ( glConfig2.dynamicLight > 0 && !glConfig2.textureFloatAvailable )
+	if ( r_dynamicLightRenderer.Get() == Util::ordinal( dynamicLightRenderer_t::TILED )
+		&& !glConfig2.textureFloatAvailable )
 	{
-		Log::Warn("Tiled dynamic light renderer not used because GL_ARB_texture_float is not available.");
-		glConfig2.dynamicLight = 0;
+		if ( glConfig2.dynamicLight || glConfig2.staticLight )
+		{
+			Log::Warn("Tiled dynamic light renderer not used because GL_ARB_texture_float is not available.");
+		}
+
+		glConfig2.dynamicLight = false;
+		glConfig2.staticLight = false;
```

Comment:
Why this `staticLight = false`?

Reply:
Those two features use the dynamic light renderer.

They're selecting what kind of lights can be rendered with the dynamic light renderer.

What can be confusing is that we have a “dynamic light renderer”, rendering lights at run time, in comparison with pre-rendered lights like lightmaps.

But the game can feature static lights from the map and dynamic lights from the game, and both can be rendered with the dynamic light renderer in game.

I don't really know what are those static lights, maybe this is an alternative to lightmaps (some code seems to be mutually exclusive with `r_precomputedLighting`).

But the code explicitely was explicitely trying to renders those static lights with the legacy renderer when `r_staticLight` was set to `1` and with the tiled renderer when set to `2`. In fact the code for both kind of lights are almost a copy paste, they just have a different source.

Reply:
Some comments in the code suggest the renderer was meant to be able to render the map like Doom3 did.

Though, this feature doesn't work (like shadows), and fixing that is outside of the scope of this PR (like shadows), but this PR makes the code easier to deal with in order to be able to possibly work on fixing the feature later (like shadows).

After this PR, enabling what kind of lights can be rendered dynapically and selecting the default dynamic light renderer will be different options.